proto/cosmos/upgrade/v1beta1/upgrade.proto

```diff
@@ -40,6 +40,39 @@ message Plan {
   // If this field is not empty, an error will be thrown.
   google.protobuf.Any upgraded_client_state = 5
       [deprecated = true, (gogoproto.moretags) = "yaml:\"upgraded_client_state\""];
+
+  // Optional: Upgrade contains additional instructions for the devops or a hypervisor.
+  // App specific instructions are handled by the `info` attribute. Here we provide
+  // information such as pre-upgrade or post-upgrade commands.
+  UpgradeInstructions Upgrade = 6;
+}
+
+message UpgradeInstructions {
+  option (gogoproto.equal)            = true;
+
+  // If not empty, a shell command to be run by the upgrade manager or a hypervisor after shutting down
+  // the app and before running a new node.
+  string pre_run = 1;
+  // If not empty, a shell command to be run by the upgrade manager or a hypervisor after shutting down
+  // the app and after running a new app.
+  string post_run = 2;
+  // List of required assets to download. This follows the cosmovisor structure.
+  // SHOULD have only one entry per platform.
+  repeated Asset assets    = 3;
+  // Description contains additional information about the upgrade process. Can reference an external resource.
+  string description = 4;
+}
+
+message Asset {
+  option (gogoproto.equal)            = true;
+
+  // Platform identifier. It's composed from OS and CPU architecture. Example: "linux/amd64"
+  string platform = 1;
+  // URL to a script or binary to download. Should be related to the UpgradeInstructions pre_run or post_run
+  // commands. If multiple files are needed, then they should be packed in a gzip archive.
+  string url = 2;
+  // Checksum is a sha256 hex encoded checksum of an artifact referenced by the URL.
+  string checksum = 3;
```

Comment:
As we use `go-getter` to download the files in cosmovisor, it automatically checks for the checksum from the download url as `./foo.sh?checksum=b7d96c89d09d9e204f5fedc4d5d55b21`. So a seperate checksum property might not be needed

Reply:
My intention was to make it universal. Proto files should not assume that we are using the go implementation or cosmovisor. We could put the checksum in the URL, but then we would need to explain it in a comment and require it when registering a new proposal.